### PR TITLE
Add WordPress provider

### PR DESCRIPTION
**Pull request by jerriep**
I have registered an application with WordPress and a callback is set up to work with the IIS Express port, so please use that.

You will also need to have a Wordpress.com account to test, so just sign up if you do not have one.

### CODE REVIEW

File: src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+namespace AspNet.Security.OAuth.WordPress {
+    public class WordPressAuthenticationHandler : OAuthAuthenticationHandler<WordPressAuthenticationOptions>
+    {
+        public WordPressAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
+        {
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationHandler.cs]: nit: formatting :smile:

File: src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            identity.AddOptionalClaim(ClaimTypes.NameIdentifier, WordPressAuthenticationHelper.GetIdentifier(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim(ClaimTypes.Name, WordPressAuthenticationHelper.GetUsername(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:wordpress:email", WordPressAuthenticationHelper.GetEmail(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:wordpress:displayname", WordPressAuthenticationHelper.GetDisplayName(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:wordpress:profileurl", WordPressAuthenticationHelper.GetProfileUrl(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:wordpress:avatarurl", WordPressAuthenticationHelper.GetAvatarUrl(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:wordpress:primaryblog", WordPressAuthenticationHelper.GetPrimaryBlog(payload), Options.ClaimsIssuer);
+
+            var notification = new OAuthAuthenticatedContext(Context, Options, Backchannel, tokens, payload)
+            {
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationHandler.cs]: nit: here too.

File: src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            [NotNull] IUrlEncoder encoder,
+            [NotNull] IOptions<SharedAuthenticationOptions> externalOptions,
+            [NotNull] IOptions<WordPressAuthenticationOptions> options,
+            ConfigureOptions<WordPressAuthenticationOptions> configureOptions = null)
+            : base(next, dataProtectionProvider, loggerFactory,
+                encoder, externalOptions, options, configureOptions) {
+        }
+
+        protected override AuthenticationHandler<WordPressAuthenticationOptions> CreateHandler()
+        {
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationMiddleware.cs]: nit: :smile:

File: src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationOptions.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            CallbackPath = new PathString(WordPressAuthenticationDefaults.CallbackPath);
+
+            AuthorizationEndpoint = WordPressAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = WordPressAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = WordPressAuthenticationDefaults.UserInformationEndpoint;
+
+            SaveTokensAsClaims = false;
+
+            // Limit by default to 'auth' scope only, otherwise too many permissions get requested
+            Scope.Add("auth");
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationOptions.cs]: :+1:

File: src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.OptionsModel;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.AspNet.Builder {
+    public static class WordPressAuthenticationExtensions {
+        public static IServiceCollection ConfigureWordPressAuthentication(
+            [NotNull] this IServiceCollection services,
+            [NotNull] Action<WordPressAuthenticationOptions> configuration)
+        {
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.WordPress/WordPressAuthenticationExtensions.cs]: Looks like ReSharper doesn't want to make your life easier :smile:

**Comment by jerriep** [comment]
Fixed formatting. Not sure WTH is going on...

File: src/AspNet.Security.OAuth.WordPress/project.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+    "dependencies": {
+        "AspNet.Security.OAuth.Extensions": { "type": "build", "version": "1.0.0-*" },
+        "Microsoft.AspNet.Authentication.OAuth": "1.0.0-*",
+        "Microsoft.Framework.NotNullAttribute.Internal": { "type": "build", "version": "1.0.0-*" }
+    },
+
+    "frameworks": {
+        "dnx451": { },
+        "dnxcore50": { }
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.WordPress/project.json]: There's a bug with dependency resolution in the latest DNX. Could you please add this workaround to make CI happy? :thought_balloon: 

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.GitHub/project.json#L27-L31

Thanks!

**Comment by PinpointTownes** [comment]
Merged: https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/commit/aba27918f737991dd233f5ee433902aa3a3b6e36

Thanks Jerrie! :+1:

**Pull request closed without merging**